libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2514,25 +2514,40 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		mstore(<pos>, <shl28>(<funSel>))
 		let <end> := <encodeArgs>(add(<pos>, 4) <argumentString>)
 
-		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <reservedReturnSize>)
+		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <staticReturndataSize>)
 		<?noTryCall>
 			if iszero(<success>) { <forwardingRevert>() }
 		</noTryCall>
 		<?+retVars> let <retVars> </+retVars>
 		if <success> {
-			<?dynamicReturnSize>
-				// copy dynamic return data out
-				returndatacopy(<pos>, 0, returndatasize())
-			</dynamicReturnSize>
+			<?isReturndataSizeDynamic>
+				<?comment>
+					// explicit returndata copy needed: we don't know the size upfront so we passed zero to the <call>
+				</comment>
```

Comment:
I'd do it like this to make it more concise:
```suggestion
				<?comment> explicit returndata copy needed: we don't know the size upfront so we passed zero to the <call> </comment>
```

Reply:
By the way, @chriseth suggested putting the comment at the top (i.e. above the `Whiskers templ(R"(` part). I think the comment is most helpful right here but the top would be a reasonable compromise so I'd be fine with that too.

If you change that, use the previous longer version of the comment that addressed both cases of course (and adjust it to make sense in the new spot).